File: C2203L/JP1/11-11-2022/EmployeeManagementProgram/src/main/java/org/example/models/Employee.java
package org.example.models;

import org.example.Utilities;

import static org.example.Utilities.getScanner;

public class Employee {
    private int id;
    private String name;
    private int workingDays;
    private Float dailySalary;

    public void input() {
        System.out.println("Enter id: ");
        this.id = getScanner().nextInt();

        System.out.println("Enter name: ");
        this.name = getScanner().next();

        while (true) {
            System.out.println("Enter working days: ");
            this.workingDays = getScanner().nextInt();
            boolean isValidWorkingDay = this.workingDays >= 1 && this.workingDays <= 31;
            if(isValidWorkingDay) {
                break;
            } else {
                System.err.println("Working days must be 1 to 31");
            }
        }

        while (true) {
            System.out.println("Enter daily salary: ");
            this.dailySalary = getScanner().nextFloat();
            boolean isValidDailySalary = this.dailySalary >= 10 && this.dailySalary <= 100;
            if(isValidDailySalary) {
                break;
            } else {
                System.err.println("Daily salary must be 10 to 100");
            }
        }


    }
    public Employee() {}
    public Employee(int id, String name, int workingDays, Float dailySalary) {
        this.id = id;
        this.name = name;
        this.workingDays = workingDays;
        this.dailySalary = dailySalary;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getWorkingDays() {
        return workingDays;
    }

    public void setWorkingDays(int workingDays) {
        this.workingDays = workingDays;
    }

    public Float getDailySalary() {
        return dailySalary;
    }

    public void setDailySalary(Float dailySalary) {
        this.dailySalary = dailySalary;
    }
    public Float getMonthlySalary() {
        return dailySalary * workingDays;
    }
    @Override
    public String toString() {
        return "Employee{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", workingDays=" + workingDays +
                ", dailySalary=" + dailySalary +
                ", MonthlySalary =" + getMonthlySalary();
    }
}

File: C2209GDoiCan/IDP/17-12-2024/employee.dart
import 'dart:io';

class Employee {
  final String id;
  final String name;
  final int workingDays;
  final int dailySalary;
  Employee(
      {required this.id,
      required this.name,
      required this.workingDays,
      required this.dailySalary});
  static Employee input() {
    String id = '';
    String name = '';
    int workingDays = 0;
    int dailySalary = 0;
    while (id.length != 6) {
      print('Enter employee\'s id : ');
      id = (stdin.readLineSync() ?? '');
      if (id.length != 6) {
        print('Employee\'s id must be 6 characters');
      }
    }
    while (name.length < 5 || name.length > 100) {
      print('Enter employee\'s name : ');
      name = (stdin.readLineSync() ?? '');
      if (name.length < 5 || name.length > 100) {
        print('Employee\'s name must be 5 to 100 characters');
      }
    }
    while (workingDays < 1 || workingDays > 31) {
      print('Enter employee\'s workingDays : ');
      name = (stdin.readLineSync() ?? '');
      if (workingDays < 1 || workingDays > 31) {
        print('Employee\'s workingDays must be 1 to 31');
      }
    }
    while (dailySalary < 10 || dailySalary > 100) {
      print('Enter employee\'s salary : ');
      name = (stdin.readLineSync() ?? '');
      if (dailySalary < 10 || dailySalary > 100) {
        print('Employee\'s dailySalary must be 10 to 100');
      }
    }
    return Employee(
        id: id, name: name, workingDays: workingDays, dailySalary: dailySalary);
  }
}
